crates/nu-command/src/filters/get.rs

```diff
@@ -139,6 +145,56 @@ If multiple cell paths are given, this will produce a list of values."#
     }
 }
 
+// the PipelineData.follow_cell_path function, when given a
+// stream, collects it into a vec before doing its job
+//
+// this is fine, since it returns a Result<Value ShellError>,
+// but if we want to follow a PipelineData into a cell path and
+// return another PipelineData, then we have to take care to
+// make sure it streams
+pub fn follow_cell_path_into_stream(
+    data: PipelineData,
+    signals: Signals,
+    cell_path: Vec<PathMember>,
+    head: Span,
+    insensitive: bool,
+) -> Result<PipelineData, ShellError> {
+    match data {
+        PipelineData::ListStream(stream, ..) => {
+            let has_int_member = cell_path.iter().any(|it| match it {
+                PathMember::String { .. } => false,
+                PathMember::Int { .. } => true,
+            });
+
+            if has_int_member {
+                // when given an integer/indexing, we fallback to
+                // the default nushell indexing behaviour
+                Value::list(stream.into_iter().collect(), head)
+                    .follow_cell_path(&cell_path, insensitive)
+                    .map(|it| it.into_pipeline_data())
+            } else {
+                let result = stream
+                    .into_iter()
+                    .map(move |value| {
+                        let span = value.span();
+
+                        match value.follow_cell_path(&cell_path, insensitive) {
+                            Ok(v) => v,
+                            Err(error) => Value::error(error, span),
+                        }
+                    })
+                    .into_pipeline_data(head, signals);
+
+                Ok(result)
+            }
+        }
```

Comment:
i think it's a bit cleaner with the conditional pulled out of the match:

```suggestion
    // when given an integer/indexing, we fallback to
    // the default nushell indexing behaviour
    let has_int_member = cell_path
        .iter()
        .any(|it| matches!(it, PathMember::Int { .. }));
    match data {
        PipelineData::ListStream(stream, ..) if !has_int_member => {
            let result = stream
                .into_iter()
                .map(move |value| {
                    let span = value.span();

                    match value.follow_cell_path(&cell_path, insensitive) {
                        Ok(v) => v,
                        Err(error) => Value::error(error, span),
                    }
                })
                .into_pipeline_data(head, signals);

            Ok(result)
        }
```